opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/aggregation.py

```diff
@@ -0,0 +1,495 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from bisect import bisect_left
+from dataclasses import replace
+from logging import getLogger
+from math import inf
+from threading import Lock
+from typing import Generic, List, Optional, Sequence, TypeVar
+
+from opentelemetry._metrics import (
+    Asynchronous,
+    Counter,
+    Histogram,
+    Instrument,
+    ObservableCounter,
+    ObservableGauge,
+    ObservableUpDownCounter,
+    Synchronous,
+    UpDownCounter,
+)
+from opentelemetry.sdk._metrics.measurement import Measurement
+from opentelemetry.sdk._metrics.point import AggregationTemporality, Gauge
+from opentelemetry.sdk._metrics.point import Histogram as HistogramPoint
+from opentelemetry.sdk._metrics.point import PointT, Sum
+from opentelemetry.util._time import _time_ns
+
+_PointVarT = TypeVar("_PointVarT", bound=PointT)
+
+_logger = getLogger(__name__)
+
+
+class _Aggregation(ABC, Generic[_PointVarT]):
+    def __init__(self):
+        self._lock = Lock()
+
+    @abstractmethod
+    def aggregate(self, measurement: Measurement) -> None:
+        pass
+
+    @abstractmethod
+    def collect(self) -> Optional[_PointVarT]:
+        pass
+
+
+class _DropAggregation(_Aggregation):
+    def aggregate(self, measurement: Measurement) -> None:
+        pass
+
+    def collect(self) -> Optional[_PointVarT]:
+        pass
+
+
+class Aggregation(ABC):
```

Comment:
Renamed `_AggregationFactory` to `Aggregation` for the base class of all public aggregations. This class should be public because it is exposed in the documentation as a type of a public argument.